boa/core/build.py

```diff
@@ -520,10 +520,54 @@ def execute_build_script(m, src_dir, env, provision_only=False):
         #     if stats is not None:
 
 
-def download_source(m):
-    # Download all the stuff that's necessary
+def _try_download(m, interactive):
+    try:
+        source.provide(m)
+    except RuntimeError as e:
+        if interactive:
+            msg = str(e)
+
+            if "mismatch: " in msg:
+                # parse the mismatch
+                parts = msg.split(" ")
+                hash_type = parts[0]
+                hash_pkg = parts[2]
+                hash_recipe = parts[4]
+                console.print(
+                    f"[red]Error: {hash_type} hash mismatch![/red]\nExpected: [red]{hash_recipe}[/red]\nGot:      [green]{hash_pkg}"
+                )
+
+                answer = Confirm.ask(
+                    "Do you want to automatically update your recipe with the new hash? [y/n]"
```

Comment:
Does this default to yes or no? It might be nice to capitalise the cosponsoring letter in `y/n`.

(I think this should default to `N` given it's a potential security issue)

Reply:
agreed, although this is only used when you run with `-i` for interactive mode.
I run into this all the time when trying to make a quick version update.

Reply:
thanks for the review, I've fixed this now!

Reply:
Thanks and yes, it's definitely a pain point for me as well!